File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organisation/event/OrganizationEditWebsiteEventDto.java
package de.helfenkannjeder.helfomat.api.organisation.event;

import de.helfenkannjeder.helfomat.core.organisation.OrganisationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditWebsiteEventDto extends OrganizationEventDto {
    private String website;

    OrganizationEditWebsiteEventDto() {
    }

    public OrganizationEditWebsiteEventDto(OrganisationId organisationId, String website) {
        super(organisationId);
        this.website = website;
    }

    public String getWebsite() {
        return website;
    }
}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditWebsiteEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditWebsiteEventDto(
    override val organizationId: OrganizationId,
    val website: String?,
    override val eventApplicable: EventApplicability
) : OrganizationEventDto {

    override fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T {
        return visitor.visit(this)
    }

}